packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -356,24 +285,45 @@ class AppBar extends StatelessWidget {
     return null;
   }
 
-  Widget _buildForSize(BuildContext context, BoxConstraints constraints) {
-    assert(constraints.maxHeight < double.INFINITY);
-    final Size size = constraints.biggest;
-    final double statusBarHeight = MediaQuery.of(context).padding.top;
+  @override
+  AppBarState createState() => new AppBarState();
+}
+
+class AppBarState extends State<AppBar> {
+  bool _hasDrawer = false;
+  bool _canPop = false;
+
+  @override
+  void dependenciesChanged() {
```

Comment:
We should probably rename `dependenciesChanged` to `didChangeDependencies` to match the style guide.